vignettes/vignettes.Rmd

```diff
@@ -0,0 +1,159 @@
+vcf <- VariantAnnotation::readVcf(vcf.file, "hg19")
+vcf
+```
+
+Simple filters could be applied to a `VCF` object to remove unwanted calls.
+```{r, warning=FALSE,message=FALSE}
+# filter out low quality calls
+hq_vcf <- vcf[rowRanges(vcf)$FILTER %in% c(".", "PASS"),]
+# somatic calls have no support in the normal
+somatic_vcf <- vcf[geno(vcf)$QUAL[,"normal.bam"] == 0,]
+# somatic loss of heterozygosity has no support in the tumour
+loh_vcf <- vcf[geno(vcf)$QUAL[,"tumour.bam"] == 0,]
+```
+
+### Converting SV calls to GRanges
+The package employs `GRanges` objects for storage of breakends. Function `breakpointRanges()` converts SVs in a VCF to GRanges with one entry per breakend. A breakend on the '+' strand indicates a break immediately after the given position, and on the '-' strand indicates a break immediately before the given position. Unpaired variants are removed at this stage.
+
+```{r}
+#converting VCF object 'vcf' to GRanges object 'gr'
+suppressPackageStartupMessages(library(StructuralVariantAnnotation))
+gr <- breakpointRanges(vcf)
+gr
+```
+
+
+### Exploring breakpoints
+
+`findBreakpointOverlaps()` and `countBreakpointOverlaps()` are functions for finding and counting overlaps between GRanges objects, equivalent of `findOverlaps()` and `countOverlaps()`. All breakends must have their partner breakend included in the GRanges. A valid overlap requires that breakends on boths sides overlap.
+
+<!-- **EXAMPLE HERE FOR GENE OVERLAP** -->
+<!-- # ```{r, warning=FALSE, message=FALSE} -->
+<!-- # library(dplyr) -->
+<!-- # library(plyranges) -->
+<!-- # region.gr <- data.frame(seqnames=c('chr12','chr12'),start=c(84350,4886681),end=c(84350,4886681),strand=c('-','+'), vcfID=c('gridss100o','gridss100h'), partner=c('gridss100h','gridss100o')) %>% as_granges() -->
+<!-- # region.gr <- setNames(region.gr, c('gridss100o','gridss100h')) -->
+<!-- # region.gr -->
+<!-- # ``` -->
+
+To demonstrate `findBreakpointOverlaps()` function, we construct a toy `GRanges` object `region.gr` using two partnered breakpoints from `gr`. The output matrix reports the index of overlapping breakpoints (`queryHits` and `subjectHits`), and overlapping distances (`sizeerror`, `localbperror` and `remotebperror`). 
+```{r}
+# event sizes must be within sizemargin
+region.gr <- gr[2:3,]
+findBreakpointOverlaps(gr, region.gr)
+```
+`countBreakpointOverlaps` reports count of overlaps for each breakpoint.
+```{r}
+suppressPackageStartupMessages(library(dplyr))
+countBreakpointOverlaps(gr, region.gr)
+```
+
+`partner()` returns partner breakend for each breakend.
+```{r}
+partner(gr)
+```
+
+
+### Converting between BEDPE files and GRanges
+
+The package supports converting GRanges objects to BEDPE files. The BEDPE format is defined by [`bedtools`](https://bedtools.readthedocs.io/en/latest/content/general-usage.html).
+
+```{r}
+#suppressPackageStartupMessages(library(stringr))
+bedpe <- breakpointgr2bedpe(gr)
```

Comment:
pairs2breakpointgr and bedpe2breakpointgr should be an overloads of the breakpointRange() function, not separate function. It should be able to take a string (filename), a connection, a data frame, or a VCF. We shouldn't have 3 functions that do the same thing.